modules/core/src/lifecycle/create-props.ts

```diff
@@ -46,22 +57,40 @@ export function createProps<T>(
   return propsInstance;
 }
 
+const MergedDefaultPropsCacheKey = '_mergedDefaultProps';
+
 // Return precalculated defaultProps and propType objects if available
 // build them if needed
-function getPropsPrototype(componentClass) {
-  const defaultProps = getOwnProperty(componentClass, '_mergedDefaultProps');
+function getPropsPrototype(componentClass, extensions?: any[]) {
+  // A string that uniquely identifies the extensions involved
+  let cacheKey = MergedDefaultPropsCacheKey;
+  if (extensions) {
+    for (const extension of extensions) {
+      const ExtensionClass = extension.constructor;
+      if (ExtensionClass) {
+        cacheKey += `:${ExtensionClass.extensionName || ExtensionClass.name}`;
+      }
+    }
+  }
+
+  const defaultProps = getOwnProperty(componentClass, cacheKey);
   if (!defaultProps) {
-    createPropsPrototypeAndTypes(componentClass);
-    return componentClass._mergedDefaultProps;
+    return (componentClass[cacheKey] = createPropsPrototypeAndTypes(
```

Comment:
Rather than using a compound cache key that includes all the extensions, would it not be possible to store the defaultProps of the extension on the Extension class and then merge them here? Are you worried about the GC overhead of doing this on every frame?

Reply:
How is caching on the Extension class different from caching on the Layer class? Don't you still need to differentiate the layer with which the extension is used?

Reply:
I'm thinking about how large the cache would grow. As more extensions are added over time the same layer with different extensions would have the defaultProps duplicated. But perhaps this is needlessly worrying about a tiny bit of memory